src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -388,6 +415,24 @@ export const selectPortfolioCryptoBalanceByAssetId = createSelector(
   (byId, assetId): string => byId[assetId]
 )
 
+export const selectPortfolioCryptoHumanBalancesByFilter = createSelector(
+  selectAssets,
+  selectPortfolioAccountBalances,
+  selectPortfolioAssetBalances,
+  selectAccountIdParamFromFilter,
+  selectAssetIdParamFromFilter,
+  (assets, accountBalances, assetBalances, accountId, assetId): string => {
+    if (accountId && assetId) {
+      return fromBaseUnit(
+        bnOrZero(accountBalances[accountId][assetId]),
+        assets[assetId].precision ?? 0
+      )
+    }
+
+    return fromBaseUnit(bnOrZero(assetBalances[assetId]), assets[assetId].precision ?? 0)
```

Comment:
this should probably guard for `if (assetId) return assetBalances[assetId]...` and return `'0'` if it's not provided

Reply:
sorry, i'm wrong, looks like `assetId` is now a required part of the filter, this is good.